docs/ingestion/ingestion-spec.md

```diff
@@ -477,35 +477,37 @@ The `indexSpec` object can include the following properties:
 |-----|-----------|-------|
 |bitmap|Compression format for bitmap indexes. Should be a JSON object with `type` set to `roaring` or `concise`. For type `roaring`, the boolean property `compressRunOnSerialization` (defaults to true) controls whether or not run-length encoding will be used when it is determined to be more space-efficient.|`{"type": "roaring"}`|
 |dimensionCompression|Compression format for dimension columns. Options are `lz4`, `lzf`, `zstd`, or `uncompressed`.|`lz4`|
-|stringDictionaryEncoding|Encoding format for string typed column value dictionaries.|`{"type":"utf8"}`|
+|stringDictionaryEncoding|Encoding format for STRING-typed column value dictionaries. The default setting `utf8` suits most use cases.<br>Example to enable front coding: `{"type":"frontCoded", "bucketSize": 4}`<br>`bucketSize` is the number of values to place in a bucket to perform delta encoding. Must be a power of 2, maximum is 128. Defaults to 4.<br>See [Front coding](#front-coding) for more information.|`{"type":"utf8"}`|
```

Comment:
hmm, thinking a bit more about this, I wonder if the additional details here would be enough if we also added the bit about being new to Druid 25.0, and instead drop all of the extra sections dedicated to `stringDictionaryEncoding`.

My reasoning is that these `indexSpec` settings provide direct control over low level segment storage details. They are primarily intended for experimentation by _the Druid developers_ to assist in determining the best default values for these settings, though i imagine advanced cluster operators also use them in the process of trying to maximally optimize a cluster.

We have to have all of this text to describe implementation details of the `frontCoded` option so that the operator can begin to understand how to correctly set the `bucketSize` parameter, and the implications of different sizes also require a decent bit of understanding of both the data and the way Druid works internally.

Did we add enough information to make an informed decision? I'm not sure if I'm a good judge of that 😅 If it is actually enough to understand reasonably well I can be convinced to leave all of these details.

Reply:
Updated.